### ISSUE TITLE
`@eslint-react/no-implicit-key` is still overly strict for ComponentProps if some fields are omitted

### ISSUE THREAD

**Issue by ulrichstark** [Rel1cx/eslint-react#1476]
This is a follow up issue to #1472.

Thank you very much for your quick fix! I unfortunately noticed that the change doesn't cover all my "false positives". I'm now down from 129 to 12 problems.

If some field is omitted from the type, the spreading of `props` is reported again.

```diff
-function PaginationItem({ ...props }: React.ComponentProps<"li">) {
+function PaginationItem({ ...props }: Omit<React.ComponentProps<"li">, "value">) {
```

Complete code:
```tsx
function PaginationItem({ ...props }: Omit<React.ComponentProps<"li">, "value">) {
  return <li data-slot="pagination-item" {...props} />;
}
```

**Comment by Rel1cx** [comment]
> I'm now down from 129 to 12 problems.

Great to hear that. In the latest [v2.12.1](https://github.com/Rel1cx/eslint-react/releases/tag/v2.12.1), I've further optimised the detection of React internally-defined keys, which should cover all such scenarios. If there are any remaining false positives, please feel free to let me know.
